frontend/src/Components/ColorModeProvider/ColorModeProvider.tsx

```diff
@@ -0,0 +1,42 @@
+import { createContext } from 'preact';
+import { useCallback, useLayoutEffect, useState } from 'preact/hooks';
+import { JSXInternal } from 'preact/src/jsx';
+
+interface ColorModeContext {
+  isDark: boolean;
+  toggleColorMode: () => void;
+}
+
+enum Mode {
```

Comment:
I trink currently we are not using enums but types i would prefere to keep it consistent

Reply:
I removed the enum for consistency. Using types here instead of enums is hardly possible, because one cannot enforce the correct type in local storage.  Thus, I am using regular strings now, which makes the code a little bit susceptible to errors.

Reply:
No longer relevant since, we local storage was removed.